utils/tests/test_model_gen.py

```diff
@@ -534,4 +543,77 @@ def test_accept_del_all(self, setup_teardown, monkeypatch, capfd):
         expected_files = setup_teardown.get('file_names')[0:5]
         assert re.match(r'^Cleanup will remove models: {.*}\s*$', out.split('\n')[0])
         assert re.match(r'^Deleted 22 models, quitting...\s*$', out.split('\n')[1])
-        assert set(files) == set(expected_files)
\ No newline at end of file
+        assert set(files) == set(expected_files)
+
+class Test_model_output:
```

Comment:
Sorry if this is a dumb question but where is this being referenced?

Reply:
It all gets called in the GH Actions when pytest is run (or when you manually run pytest) - the class is to help cluster together similar tests (in this case all the tests of the actual output files), and also to use the fixture to run at the start of the tests, yield to running the tests part way through once the files exist, and then once the tests are completed it finishes running and removes all the files it created as we have to use the actual models folder so can't use a temp dir that will get dropped anwyay. 

The slight difference is this doesn't do anything to explicitely increase tests coverage because we're only looking at coverage of lines in the functions file - this tests the main script as black box based on the output, but doesn't refer to any specific part of that script to test individual parts.

That make sense?

Reply:
Yep, makes sense. Thanks for the additional info!